baseapp/abci_utils_test.go

```diff
@@ -274,6 +275,26 @@ func (s *ABCIUtilsTestSuite) TestValidateVoteExtensionsTwoVotesNilAbsent() {
 	s.Require().Error(baseapp.ValidateVoteExtensions(s.ctx, s.valStore, 3, chainID, llc))
 }
 
+func (s *ABCIUtilsTestSuite) TestDefaultProposalHandler_NoOpMempoolTxSelection() {
+	ph := baseapp.NewDefaultProposalHandler(mempool.NoOpMempool{}, nil)
+	handler := ph.PrepareProposalHandler()
+
+	// Request PrepareProposal with 5 txs, 5 bytes each, with a max size of 15.
+	// We should only select the first 3 txs.
+	resp, err := handler(s.ctx, &abci.RequestPrepareProposal{
+		Txs: [][]byte{
+			{0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+			{0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+			{0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+			{0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+			{0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+		},
+		MaxTxBytes: 15,
```

Comment:
Let's also write another test in which we specify maxBlockGas.

Reply:
We won't test anything here. See https://github.com/cosmos/cosmos-sdk/pull/17769#discussion_r1327613963